src/Mvc/Mvc.DataAnnotations/test/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -1214,6 +1222,189 @@ public void CreateValidationMetadata_SuppressRequiredInference_Noops()
             Assert.DoesNotContain(context.ValidationMetadata.ValidatorMetadata, m => m is RequiredAttribute);
         }
 
+        [Theory]
+        [InlineData(nameof(DerivedTypeWithAllNonNullProperties.Property1))]
+        [InlineData(nameof(DerivedTypeWithAllNonNullProperties.Property2))]
+        public void CreateValidationMetadata_InfersRequiredAttributeOnDerivedType_BaseAnDerivedTypHaveAllNonNullProperties(string propertyName)
+        {
+            // Arrange
+            var provider = CreateProvider();
+
+            var modelType = typeof(DerivedTypeWithAllNonNullProperties);
+            var property = modelType.GetProperty(propertyName);
+            var key = ModelMetadataIdentity.ForProperty(property, property.PropertyType, modelType);
+            var context = new ValidationMetadataProviderContext(key, ModelAttributes.GetAttributesForProperty(modelType, property));
+
+            // This test verifies how MVC reads the NullableContextOptions. We expect the property to not have a Nullable attribute on, and for
+            // the types to have NullableContext. We;ll encode our expectations as assertions so that we can catch if or when the compiler changes
```

Comment:
```suggestion
            // the types to have NullableContext. We'll encode our expectations as assertions so that we can catch if or when the compiler changes
```